### ISSUE TITLE
`TypeError: can not serialize 'TransformerData' object` when using `doc.to_bytes()` with `en_core_web_trf`

### ISSUE THREAD

**Issue by ducalpha** [explosion/spaCy#6672]
## How to reproduce the behaviour

```
import spacy
sentence = "I love you."
nlp = spacy.load('en_core_web_trf')
doc = nlp(sentence)
doc.to_bytes()
```

Error:
```
~/anaconda3/envs/dummy/lib/python3.8/site-packages/spacy/tokens/doc.pyx in spacy.tokens.doc.Doc.to_bytes()

~/anaconda3/envs/dummy/lib/python3.8/site-packages/spacy/tokens/doc.pyx in spacy.tokens.doc.Doc.to_dict()

~/anaconda3/envs/dummy/lib/python3.8/site-packages/spacy/util.py in to_dict(getters, exclude)
   1044         # Split to support file names like meta.json
   1045         if key.split(".")[0] not in exclude:
-> 1046             serialized[key] = getter()
   1047     return serialized
   1048 

~/anaconda3/envs/dummy/lib/python3.8/site-packages/spacy/tokens/doc.pyx in spacy.tokens.doc.Doc.to_dict.lambda17()

~/anaconda3/envs/dummy/lib/python3.8/site-packages/srsly/_msgpack_api.py in msgpack_dumps(data)
     11     RETURNS (bytes): The serialized bytes.
     12     """
---> 13     return msgpack.dumps(data, use_bin_type=True)
     14 
     15 

~/anaconda3/envs/dummy/lib/python3.8/site-packages/srsly/msgpack/__init__.py in packb(o, **kwargs)
     42     Pack an object and return the packed bytes.
     43     """
---> 44     return Packer(**kwargs).pack(o)
     45
     46 

~/anaconda3/envs/dummy/lib/python3.8/site-packages/srsly/msgpack/_packer.pyx in srsly.msgpack._packer.Packer.pack()

~/anaconda3/envs/dummy/lib/python3.8/site-packages/srsly/msgpack/_packer.pyx in srsly.msgpack._packer.Packer.pack()

~/anaconda3/envs/dummy/lib/python3.8/site-packages/srsly/msgpack/_packer.pyx in srsly.msgpack._packer.Packer.pack()

~/anaconda3/envs/dummy/lib/python3.8/site-packages/srsly/msgpack/_packer.pyx in srsly.msgpack._packer.Packer._pack()

~/anaconda3/envs/dummy/lib/python3.8/site-packages/srsly/msgpack/_packer.pyx in srsly.msgpack._packer.Packer._pack()

TypeError: can not serialize 'TransformerData' object
```
## Your Environment

* Operating System: Ubuntu 18.04
* Python Version Used: 3.8
* spaCy Version Used: 3.0.0rc2
* Environment Information:

## My workaround
Exclude `user_data_values` which causes the problem (and `user_data_keys`, otherwise, I could not recover the data using `Doc.from_bytes()`.
```
doc.to_bytes(exclude=['user_data_keys', 'user_data_values'])
```

BTW, what does "user_data_values" store in this case?

**Comment by adrianeboyd** [comment]
I'm not sure why the keys/values are split up in the exclude values, maybe for historical reasons. I'd suggest this instead, which covers both:

```python
doc.to_bytes(exclude=["user_data"])
```

In the future I think we'll want to support msgpack serialization for our dataclasses, but it's unlikely v3.0 will support it initially.

**Comment by ines** [comment]
Forgot to update, this should be fixed in v3.0 – we've added a custom hook to `msgpack` that handles serializing and deserializing `TransformerData`.
